src/zx/controller.rs

```diff
@@ -123,13 +124,15 @@ impl ZXController {
             ZXMachine::Sinclair128K => {
                 let mut rom0 = Vec::new();
                 let mut rom1 = Vec::new();
-                File::open(path).ok().expect("[ERROR] ROM0 not found").read_to_end(&mut rom0)
-                                                                      .unwrap();
-                let mut second_path = path.to_owned();
-                second_path.pop();
-                second_path.push('1');
+                if !path.as_ref().extension().map_or(false, |e| e == "0") {
+                    println!("[Warning] ROM0 filename should end with .0");
+                }
+                File::open(path.as_ref()).ok().expect("[ERROR] ROM0 not found").read_to_end(&mut rom0)
+                    .unwrap();
+                let mut second_path: PathBuf = path.as_ref().to_path_buf();
+                second_path.set_extension("1");
```

Comment:
Slightly changed semantics here:

- before: second ROM page is in file with _last filename character_ replaced with "1"
- after: second ROM image is in file with _extension_ replaced with "0"; also added warning

I think it's ok.

I tested it with ROMs supplied with Rustzx. It works with double extensions like `.rom.0`.